skfp/fingerprints/weiner_index.py

```diff
@@ -0,0 +1,94 @@
+from collections.abc import Sequence
+from typing import Optional, Union
+
+import numpy as np
+from rdkit.Chem import GetDistanceMatrix, Mol
+from scipy.sparse import csr_array
+
+from skfp.bases import BaseFingerprintTransformer
+from skfp.utils import ensure_mols
+
+
+class WeinerIndexFingerprint(BaseFingerprintTransformer):
+    """
+    Weiner Index fingerprint.
+
```

Comment:
This shouldn't be a fingerprint, this is a single descriptor. It is, however, a part of other descriptors, and a useful thing to have.

Create `descriptors` directory next to `fingerprints`, and maybe `topological.py` file there, and put Wiener index in there. Also, it should be a function `wiener_index(mol, distance_matrix: Optional[np.ndarray] = None) -> int`. If user has a precomputed distance matrix, it should be reused.

Reply:
Also, add the appropriate citation. See other fingerprints for a format.